rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,134 @@
+    .getOrElse("default")
+
+  /**
+   * This uses blocking IO, and so should only be used to read configuration at startup.
+   */
+  private def readConfigVarFromFilesystem(path: String, name: String): Option[String] = {
+    val file = Paths.get(path)
+    if (Files.exists(file)) {
+      try {
+        Some(new String(Files.readAllBytes(file), "utf-8"))
+      } catch {
+        case NonFatal(e) =>
+          log.error(e, "Error reading {} from {}", name, path)
+          None
+      }
+    } else {
+      log.warning("Unable to read {} from {} because it doesn't exist.", name, path)
+      None
+    }
+  }
+
+  private val clientSslContext: HttpsConnectionContext = ConnectionContext.httpsClient(sslContext)
+
+  private lazy val sslContext = {
+    val certificates = PemManagersProvider.loadCertificates(k8sSettings.apiCaPath)
+    val factory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+    val keyStore = KeyStore.getInstance("PKCS12")
+    keyStore.load(null)
+    factory.init(keyStore, Array.empty)
+    val km: Array[KeyManager] = factory.getKeyManagers
+    val tm: Array[TrustManager] =
+      PemManagersProvider.buildTrustManagers(certificates)
+    val random: SecureRandom = new SecureRandom
+    val sslContext = SSLContext.getInstance("TLSv1.2")
+    sslContext.init(km, tm, random)
+    sslContext
+  }
+
+  implicit val orderByAgeDesc: Ordering[Member] = Member.ageOrdering
+  def receive = onMessage(0, SortedSet.empty)
+
+  private def onMessage(deletionCost: Int, membersByAgeDesc: SortedSet[Member]): Receive = {
+    case ClusterEvent.MemberUp(m) =>
+      log.debug("Received MemberUp {}", m)
+      updateDeletionCost(deletionCost, membersByAgeDesc + m)
+    case ClusterEvent.MemberRemoved(m, _) =>
+      log.debug("Received MemberRemoved {}", m)
+      updateDeletionCost(deletionCost, membersByAgeDesc - m)
+    case _ =>
+  }
+
+  private def updateDeletionCost(
+      existingCost: Int,
+      membersByAgeDesc: immutable.SortedSet[Member],
+      costStrategy: CostStrategy = OlderCostsMore): Unit = {
+    val newCost: Int = costStrategy.costOf(cluster.selfMember, membersByAgeDesc).getOrElse(0)
+    log.debug(
+      "Calculated cost={} for member={} in members by age (desc): {}",
+      newCost,
+      cluster.selfMember,
+      membersByAgeDesc)
+
+    if (newCost != existingCost) {
+      log.debug(
+        "Updating pod-deletion-cost annotation for pod: [{}] with cost: [{}] (previously: {}). Namespace: [{}]",
+        k8sSettings.podName,
+        newCost,
+        existingCost,
+        podNamespace
+      )
+
+      val request = ApiRequests.podDeletionCost(k8sSettings, apiToken, podNamespace, newCost)
+
+      val responseFuture: Future[HttpResponse] =
+        if (k8sSettings.secure)
+          http.singleRequest(request, clientSslContext)
+        else
+          http.singleRequest(request)
+
+      val res = Await.result(responseFuture, 2.seconds)
```

Comment:
Don't use blocking here, you can pipe the result of the future back to `self` https://doc.akka.io/docs/akka/current/actors.html#ask-send-and-receive-future

I think it can become the new state immediately without waiting for the response of the kubernetes request.

If the kubernetes request fails you can try again, after a scheduled backoff delay